### ISSUE TITLE
Dependency resolution not working

### ISSUE THREAD

**Issue by corneil** [melix/jmh-gradle-plugin#108]
When using the following type of rule or `io.spring.dependency-management` the versions are not resolved when jmh executes.

```
configurations.all {
    resolutionStrategy.eachDependency { DependencyResolveDetails details ->
        if (details.requested.group == 'org.jetbrains.kotlin') {
            details.useVersion kotlinVersion
        }
    }
}
```

I believe it is the detached configuration not allowing Gradle to apply resolutionStrategy or the dependency-management to resolve versions.

**Comment by corneil** [comment]
I the runtime configuration is named and not hidden the resolution works as expected,

**Comment by corneil** [comment]
By changing the method `createJmhRuntimeConfiguration` in `JMHPlugin.groovy` the gradle resolutionStrategy has an opportunity to add version information.

```groovy
@CompileStatic
    private static Configuration createJmhRuntimeConfiguration(Project project, JMHPluginExtension extension) {
        try {
            project.configurations.create('jmhRuntime')
        } catch (Exception x) {
            // ignore
        }
        project.afterEvaluate {
            def newConfig = project.configurations.getByName('jmhRuntime')
            newConfig.dependencies.addAll(project.configurations.getByName('jmh').allDependencies)
            newConfig.dependencies.addAll(project.configurations.getByName('runtime').allDependencies)
            if (extension.includeTests) {
                newConfig.dependencies.addAll(
                        project.configurations.getByName('testRuntime').allDependencies)
            }
        }
        newConfig
    }
```

**Comment by RezaLabude** [comment]
Hello
we have the same problem with the io.spring.dependency-management plugin when it is used with jmh. As pointed out here https://github.com/spring-gradle-plugins/dependency-management-plugin/issues/183 this can only be resolved in the JMH plugin and the propsed solution seems to me viable. Why must this new configuration be unnamed and hidden anyway? 
with older version of the jmh plugin it was possible to work with the io.spring.dependency-management plugin
i hope you can provide a solution for this problem.
best regards

**Comment by melix** [comment]
Right, it's a problem in Gradle: resolution strategies are not applied to detached configurations. I'm reluctant on making this a hard configuration since it's an internal implementation detail. Let me see what I can do...

**Comment by RezaLabude** [comment]
I'm no expert on gradle at all, but perhaps this idea might help you:
newConfig.setCanBeResolved(true)
or perhaps you need to do something with extendsFrom:
newCconfig = project.configurations.detachedConfiguration().extendsFrom(project.configurations.getByName('jmhRuntime'))
I took all of this from this thread:
https://github.com/ben-manes/gradle-versions-plugin/issues/127
I hope this helps you somehow

**Comment by melix** [comment]
No, `setCanBeResolved` has nothing to do with the problem. The problem is that detached configurations (configurations without a container, used for internal resolution) don't get any resolution rule, because those are precisely attached to the container. There are ways to make them happen on a detached configurations, but it involves a bit of trickery using reflection and internal APIs.

In the meantime it may be a good reason to make it a hard configuration.

**Comment by melix** [comment]
Fix is available in 0.4.5-beta-2 (requires Gradle 4.3-rc-2)
